aptos-move/move-examples/nft_auction_house/sources/auction_house_own_listing.move

```diff
@@ -0,0 +1,531 @@
+            min_price,
+            start_sec,
+            expiration_sec,
+            min_incremental_price,
+        );
+    }
+
+    public fun create_auction_with_token_id<CoinType>(
+        owner: &signer,
+        token_id: TokenId,
+        amount: u64,
+        min_price: u64,
+        start_sec: u64, // specify when the auction starts
+        expiration_sec: u64, // specify when the auction ends
+        min_incremental_price: u64,
+    ): u64 acquires Auctions {
+        let auction = generate_auction_data<CoinType>(
+            owner,
+            token_id,
+            amount,
+            min_price,
+            start_sec,
+            expiration_sec,
+            expiration_sec + AUCTION_DELAY_TIME_SEC, // allow time to withdraw
+            min_incremental_price,
+        );
+
+        // initialized coin store when listing
+        if (!coin::is_account_registered<CoinType>(signer::address_of(owner))) {
+            coin::register<CoinType>(owner);
+        };
+
+        let auctions = borrow_global_mut<Auctions<CoinType>>(@auction_house);
+        event::emit_event<ListingEvent>(
+            &mut auctions.listing_event,
+            listing::create_listing_event(
+                listing::get_listing_id(&auction.listing),
+                token_id,
+                amount,
+                min_price,
+                false,
+                start_sec,
+                expiration_sec,
+                @auction_house,
+            ),
+        );
+
+        let next_id = auctions.cur_auction_id + 1;
+        *(&mut auctions.cur_auction_id) = next_id;
+        table::add(&mut auctions.all_active_auctions, next_id, auction);
+        next_id
+    }
+
+    public entry fun bid<CoinType>(
+        bidder: &signer,
+        creator: address,
+        collection_name: String,
+        token_name: String,
+        property_version: u64,
+        token_amount:u64,
+        offer_price: u64,
+        auction_id: u64,
+    ) acquires Auctions {
+        // create bid and store it under the user account
+        let token_id = token::create_token_id_raw(creator, collection_name, token_name, property_version);
+        create_bid_with_token_id<CoinType>(bidder, token_id, token_amount, offer_price, auction_id);
+    }
+
+    public fun create_bid_with_token_id<CoinType>(
+        bidder: &signer,
+        token_id: TokenId,
+        token_amount:u64,
+        offer_price: u64,
+        auction_id: u64,
+    ): ID acquires Auctions {
+        let auctions = borrow_global_mut<Auctions<CoinType>>(@auction_house);
+        assert!(table::contains(&mut auctions.all_active_auctions, auction_id), error::not_found(EAUCTION_NOT_EXIST));
+        let auction = table::borrow_mut(&mut auctions.all_active_auctions, auction_id);
+        // min_incremental_price should be bigger than 0, this also ensure the offer_price is bigger than 0
+        assert!(auction.min_incremental_price > 0, error::invalid_argument(EBID_MIN_INCREMENTAL_IS_ZERO));
```

Comment:
Why is this check here and not in auction creation?

Reply:
Good suggestion, moved this check to auction creation